Setting-Up-Ubuntu-For-Programming.md

```diff
@@ -0,0 +1,219 @@
+- [The Art of Command Line](https://github.com/jlevy/the-art-of-command-line)
+- [Learn Enough Command Line to Be Dangerous](https://www.learnenough.com/command-line-tutorial)
+
+## Installing DevTools and modern web browsers
+
+#### Git
+```bash
+sudo apt-get install git
+```
+
+#### Node JS
+```bash
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.31.0/install.sh | bash
+nvm install 5.0
+```
+
+#### Ruby
+```bash
+curl -sSL https://get.rvm.io | bash -s stable
+rvm install 2.3.1
+```
+
+#### Python
+```bash
+curl -L https://raw.githubusercontent.com/yyuu/pyenv-installer/master/bin/pyenv-installer | bash
+```
+
+#### Modern Web Browsers
+
+The most popular web browsers used by developers are [Chrome](https://www.google.com/chrome/)/[Chromium](https://www.chromium.org/) and [Firefox Developer Edition](https://www.mozilla.org/firefox/developer/).
+
+**Installing Chromium:**
+```bash
+sudo apt-get install chromium-browser
+```
+
+**Installing Firefox Developer Edition:**
+```bash
+sudo add-apt-repository ppa:ubuntu-mozilla-daily/firefox-aurora
+sudo apt-get update
+sudo apt-get install firefox
+```
+
+## Downloading Text Editors and IDEs
+
+#### Text Editors
+
+The most popular text editors right now are [Atom](https://atom.io/) by GitHub and [Sublime Text](https://www.sublimetext.com/).
+
+**Installing Atom:**
+```bash
+sudo add-apt-repository ppa:webupd8team/atom
+sudo apt-get update
+sudo apt-get install atom
+```
+
+**Installing Sublime Text 3:**
+```bash
+sudo add-apt-repository ppa:webupd8team/sublime-text-3
+sudo apt-get update
+sudo apt-get install sublime-text-installer
+```
+
+#### IDEs
+
+**Installing Eclipse:**
+
+```bash
+sudo apt-get install openjdk-7-jdk
+```
+
+Download the latest Eclipse package: http://www.eclipse.org/downloads/?osType=linux
+
+```bash
+sudo mv eclipse-inst-linux64.tar.gz /opt/ && cd /opt
+sudo tar -xvf eclipse-inst-linux64.tar.gz
+```
+
+Create a new file called `eclipse.desktop` in the `/usr/share/applications/` directory and add the lines below:
+
+```
+[Desktop Entry]
+Name=Eclipse 
+Type=Application
+Exec=/opt/eclipse/eclipse
+Terminal=false
+Icon=/opt/eclipse/icon.xpm
+Comment=Integrated Development Environment
+NoDisplay=false
+Categories=Development;IDE
+Name[en]=eclipse.desktop
+```
+
+You can now drag this file to the launcher, enabling you to start Eclipse.
+
+**Installing Intellij IDEA:**
+
+Download the latest version of Intellij IDEA: https://www.jetbrains.com/idea/download/#section=linux
+
+```bash
+sudo mv idea-2016.1.1.tar.gz /opt/ && cd /opt
+sudo tar -xvf idea-2016.1.1.tar.gz
+```
+
+Create a new file called `intellij-idea.desktop` in the `/usr/share/applications/` directory and add the below lines:
+
+```
+[Desktop Entry]
+Name=IntelliJ IDEA
+Type=Application
+Exec=/opt/idea/bin/idea.sh
+Terminal=false
+Icon=/opt/idea/bin/idea.png
+Comment=Integrated Development Environment
+NoDisplay=false
+Categories=Development;IDE;
+Name[en]=IntelliJ IDEA
+Name[en_US]=IntelliJ IDEA
+```
```

Comment:
No need for manual desktop launcher creation. IntelliJ asks to creates them by default on first run and later can be created by clicking a button.